part_1/solution.py

```diff
@@ -0,0 +1,1327 @@
+    RandGaussianNoised(keys=source_channel, prob=0.5, mean=0.0, std=0.3),
+    RandGaussianSmoothd(
+        keys=source_channel,
+        sigma_x=(0.25, 0.75),
+        sigma_y=(0.25, 0.75),
+        sigma_z=(0.0, 0.0),
+        prob=0.5,
+    ),
+]
+
+normalizations = [
+    NormalizeSampled(
+        keys=source_channel + target_channel,
+        level="fov_statistics",
+        subtrahend="mean",
+        divisor="std",
+    )
+]
+
+data_module.augmentations = augmentations
+data_module.setup("fit")
+
+# get the new data loader with augmentation turned on
+augmented_train_dataloader = data_module.train_dataloader()
+
+# Draw batches and write to tensorboard
+writer = SummaryWriter(log_dir=f"{log_dir}/view_batch")
+augmented_batch = next(iter(augmented_train_dataloader))
+log_batch_tensorboard(augmented_batch, 0, writer, "augmentation/some")
+writer.close()
+
+
+# %% [markdown]
+# Visualize directly on Jupyter
+
+# %%
+log_batch_jupyter(augmented_batch)
+
+# %% [markdown]
+"""
+## Train a 2D U-Net model to predict nuclei and membrane from phase.
+
+### Construct a 2D UNeXt2 using VisCy
+See ``viscy.unet.networks.Unet2D.Unet2d`` ([source code](https://github.com/mehta-lab/VisCy/blob/7c5e4c1d68e70163cf514d22c475da8ea7dc3a88/viscy/unet/networks/Unet2D.py#L7)) for configuration details.
+"""
+# %%
+# Create a 2D UNet.
+GPU_ID = 0
+
+BATCH_SIZE = 12
+YX_PATCH_SIZE = (256, 256)
+
+# Dictionary that specifies key parameters of the model.
+
+phase2fluor_config = dict(
+    in_channels=1,
+    out_channels=2,
+    encoder_blocks=[3, 3, 9, 3],
+    dims=[96, 192, 384, 768],
+    decoder_conv_blocks=2,
+    stem_kernel_size=(1, 2, 2),
+    in_stack_depth=1,
+    pretraining=False,
+)
+
+phase2fluor_model = VSUNet(
+    architecture="UNeXt2_2D",  # 2D UNeXt2 architecture
+    model_config=phase2fluor_config.copy(),
+    loss_function=MixedLoss(l1_alpha=0.5, l2_alpha=0.0, ms_dssim_alpha=0.5),
+    schedule="WarmupCosine",
+    lr=2e-6,
+    log_batches_per_epoch=5,  # Number of samples from each batch to log to tensorboard.
+    freeze_encoder=False,
+)
+
+# %% [markdown]
+"""
+### Instantiate data module and trainer, test that we are setup to launch training.
+"""
+# %%
+source_channel = ["Phase3D"]
+target_channel = ["Nucl", "Mem"]
+# Setup the data module.
+phase2fluor_2D_data = HCSDataModule(
+    data_path,
+    architecture="UNeXt2_2D",
+    source_channel=source_channel,
+    target_channel=target_channel,
+    z_window_size=1,
+    split_ratio=0.8,
+    batch_size=BATCH_SIZE,
+    num_workers=8,
+    yx_patch_size=YX_PATCH_SIZE,
+    augmentations=augmentations,
+    normalizations=normalizations,
+)
+phase2fluor_2D_data.setup("fit")
+# fast_dev_run runs a single batch of data through the model to check for errors.
+trainer = VSTrainer(accelerator="gpu", devices=[GPU_ID], fast_dev_run=True)
+
+# trainer class takes the model and the data module as inputs.
+trainer.fit(phase2fluor_model, datamodule=phase2fluor_2D_data)
```

Comment:
I was confused when I first ran this and thought it was already a full training loop here. Is this a test of whether the models work? If yes, I think Lightning already does a sanity check of sorts.

Reply:
I think we do this because the sanity check only runs the validation. Happy to remove it if we think is redundant